faunadb-java-dsl/src/main/java/com/faunadb/client/types/FaunaField.java

```diff
@@ -0,0 +1,36 @@
+package com.faunadb.client.types;
+
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+/**
+ * <p>Maps a field/getter or constructor parameter to FaunaDB object property while encoding/decoding an object.</p>
```

Comment:
"Maps a field, getter, or constructor parameter to an object field in FaunaDB when encoding or decoding an object".

I don't really like the term "object field" or "object property" so if someone has a better idea...